File: VivekFileShare1/main.py
# This is a sample Python script

# Import dependencies
import random
import matplotlib.pyplot as plt
import json
import os

# Print to console/stdout
print("Hello, {0}!".format(os.environ['DOMINO_PROJECT_OWNER']))


# Define a helper function to generate a random number
def random_number(start, stop):
    return random.uniform(start, stop)


# Plot the values of random points
x = random.sample(range(1000), 100)
xbins = [0, len(x)]
plt.bar(range(0, 100), x)
plt.show()
plt.savefig('results/myHistogramFromPython.png', format='png')

# Generate and save some key statistics to dominostats.json
# Learn more at https://docs.dominodatalab.com/en/latest/reference/runs/Diagnostic_statistics_with_dominostats.json.html
r2 = round(random_number(0, 1), 4)
p = round(random_number(0, 1), 4)
with open('dominostats.json', 'w') as f:
    f.write(json.dumps({"R^2": r2, "p-value": p}))

File: VivekFileShare1/main.R
# This is a sample R script

# Load dependencies
library(jsonlite)

# Plot a histogram of 100 random numbers
# In R code, Domino automatically saves your plots to svg files in your project's "results" folder
hist(rnorm(100))

# Generate and save some key statistics to dominostats.json
# Learn more at https://docs.dominodatalab.com/en/latest/reference/runs/Diagnostic_statistics_with_dominostats.json.html
r2 <- runif(1)
p <- runif(1)
diagnostics = list("R^2" = r2, "p-value" = p)
fileConn<-file("dominostats.json")
writeLines(toJSON(diagnostics), fileConn)
close(fileConn)
